packages/ragbits-core/src/ragbits/core/prompt/prompt.py

```diff
@@ -119,14 +130,17 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             self._render_template(self.system_prompt_template, input_data) if self.system_prompt_template else None
         )
         self.rendered_user_prompt = self._render_template(self.user_prompt_template, input_data)
+        self._image_input_type = InputType(collection_type=list, element_type=bytes)
+        self.image_fields: list[str] = []
+        self.images = self._get_images_from_input_data(input_data)
 
         # Additional few shot examples that can be added dynamically using methods
         # (in opposite to the static `few_shots` attribute which is defined in the class)
         self._instace_few_shots: list[FewShotExample[InputT, OutputT]] = []
         super().__init__()
 
     @property
-    def chat(self) -> ChatFormat:
+    def chat(self) -> ChatFormat | ChatFormatWithImages:
```

Comment:
Why? Is the only reason the fact that I insisted that having a precise typing for `ChatFormat` would be a good thing, or is there another reason?

Having a method that _sometimes_ returns a data structure that is ready to be used with LLMs, and _sometimes_ returns an anonymous tuple that needs to be processed (and will fail when given directly to the LLM) makes the API much, much more confusing. If I were to choose between an imprecise type annotation and this, I would definitely choose an imprecise type annotation.

Reply:
`ChatFormatWithImages` has been removed - even two lines of # type: ignore were enough to overcome issue with definition of `ChatFormat`